lib/rouge/lexers/perl6pod.rb

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Pod6 < RegexLexer
+      title "Perl 6 Pod"
+      desc "Perl 6 Plain' Ol Documentation (perl6.org)"
+
+      tag 'perl6pod'
+      aliases 'pod6'
+
+      filenames '*.pod', '*.pod6'
+
+      state :root do
+         rule(/\=begin pod/m, Keyword, :pod6)
+         rule(/^#.*$/, Comment)
+      end
+
+      state :pod6 do
+         rule(/\=end pod/m, Keyword, :pop!)
+
+         rule(/\=NAME/, Keyword, :semantic)
```

Comment:
Please merge the ones that are logically the same to a single regex. Rouge runs the regex matches one after another, so generally its better to avoid an unnecessary proliferation of regexes for things that can be easily combined..